### DISCUSSION TITLE
Anybody get graphql subscriptions working?

### DISCUSSION THREAD

**Discussion by tettoffensive** [vercel/swr#636] [General]
I'm using Next.js with my `/pages/api/graphql` running `apollo-server-micro`. On the client, I've been using `swr`. I'm just needing to get notified when a new comment is added by another user. So I thought I'd add a graphql subscription. The problem is, I'm not sure how to get the ApolloServer to handle these requests. I've seen numerous documentation that the client side should be doing something like this:

```ts
const client = new SubscriptionClient('ws://localhost:3000/api/graphql', {
    reconnect: true,
    connectionParams: {
      headers,
    },
  });
```

But I haven't been able to make a connection to the server because I guess I need to setup a websocket somehow or maybe there's an alternative?

```ts
const apolloServer = new ApolloServer({
  typeDefs,
  resolvers,
  subscriptions: {
    onConnect: () => console.log('connected'),
    onDisconnect: () => console.log('disconnected'),
  },
});
```

**Comment by JayMGurav** [comment]
Found a workaround to use subscriptions on both client as well as server-side, you refer it [here](https://github.com/JayMGurav/DevNews/blob/master/pages/api/graphql.js). But this solution has its own issues 

- hot-reloading server code doesn't work any more.
- even if subscriptions(socket connection) works in my development env, I'm not able to get them connected in vercel production  deployment and this is because [Vercel does not support websockets](https://vercel.com/docs/platform/limits#websockets)
